src/components/templates/Auditorium/Auditorium.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState } from "react";
+
+import { SectionPreviews } from "./components/SectionPreviews";
+import { Section } from "./components/Section";
+
+import { AuditoriumVenue } from "types/venues";
+
+const enum AuditoriumVideoPosition {
+  TOP = "top",
+  CENTER = "center",
+}
+
+export interface AuditoriumProps {
+  venue: AuditoriumVenue;
+}
+
+/*
+  TODO: Sections will be stored as a collection of a Auditorium venue.
+  Each should have its own autogenerated id.
+  Each will contain chats.
+
+  I guess we should have more stricter view choice. For example: 6, 10, 20, 30+ sections.
+  This way we can provide best UI for each(most used) case
+*/
```

Comment:
> If this is correct, how would you imagine the 'world chat' to be laid out, out of curiosity? Mostly wanting to ensure our choices here can align/make sense across all chat things, and don't end up with us 'fragmenting' chats across too many disparate places.

These chats are `section` chats. They are not related to world chat.

>  we also need to pay attention to/manually handle when their 'parent collection item' is deleted, as they don't automatically get deleted together.

 I guess, when the venue is deleted, they should be deleted as well
 

>  What sort of optimisations do you feel like we could/would do based on this stricter set of choices? And do you feel like we need to arbitrarily restrict them to be able to make those choices?

I'm not 100% sure about it and probably will be able to tell more, when I start implementing that component

Reply:
> These chats are section chats. They are not related to world chat.

@mike-lvov Missed the nuance of my question entirely.

Currently we have 1 chat collection, possible 2 (not sure how venue/private chats are laid out) in a certain 'structure within our firestore database. We have planned to add at least 2 more chat types (this one, and world chat), which are likely going to end up in 2 more collections within the database.

I want to know where you expect all of our chat collections to end up, and whether there is a clean/simple/useful way that we can 'batch them all together' (if that will make sense to do) in some logical path like:

- `chat/world/:worldId/:chatId`
- `chat/venue/:venueId/:chatId`
- `chat/section/:sectionId/:chatId`
- `chat/private/:userId/:chatId`

Rather than being 'scattered all over the place' like:

- `venues/:venueId/chats/:chatId`
- `privatechats/:userId/chats/:chatId`
- etc

So, I ask the question again:

> how would you imagine the 'world chat' to be laid out, out of curiosity? Mostly wanting to ensure our choices here can align/make sense across all chat things, and don't end up with us 'fragmenting' chats across too many disparate places.

Reply:
> I guess, when the venue is deleted, they should be deleted as well

@mike-lvov That would probably generally be my thoughts as well.

---

> > What sort of optimisations do you feel like we could/would do based on this stricter set of choices? And do you feel like we need to arbitrarily restrict them to be able to make those choices?
> 
> I'm not 100% sure about it and probably will be able to tell more, when I start implementing that component

@mike-lvov Makes sense. Once you get to that aspect of things, would be good to raise your learnings/thoughts/etc for discussion, so we can see if we want/need to create these 'arbitrarily sized restrictions'

Reply:
`whether there is a clean/simple/useful way that we can 'batch them all together'`. TBH, I don't think they should be located in one place. Since I don't imagine a situation, where we would need to fetch all chats together, I don't see a need to store them in a separate collection

this is somewhat related https://firebase.googleblog.com/2019/06/understanding-collection-group-queries.html

Reply:
> TBH, I don't think they should be located in one place. Since I don't imagine a situation, where we would need to fetch all chats together, I don't see a need to store them in a separate collection

Yeah, that's fair enough. I guess my question/thinking here was whether we want to think about chats as a 'first class' concept (and modelling them from that 'single collection view' as described above), or if they are just an 'aspect' of other things (in which case we would continue to scatter them across the various 'models' they relate to)

At the end of the day, I expect either pattern will work fine. It's probably just a case of 'database design'/etc.

---

RE: collection group queries

Yeah, seen that before (possibly when you linked it in the past?). It's pretty neat. I think that would let us query 'all chats' (if we ever needed to), though possibly (not 100%) would need to ensure they are all at the same 'level' in the hierarchy to do so (eg. all at something like `foo/*/chats`)

This is probably the most relevant chunk for now IMO:

- https://firebase.googleblog.com/2019/06/understanding-collection-group-queries.html#imgFull:~:text=It%20still%20seems%20like%20I%20could,How%20are%20collection%20group%20queries%20better%3F

And this is probably reason enough on it's own to stick to nested documents:

> Also, from a security rules standpoint, it's fairly common to restrict child documents based on some data that exists in their parent, and that's significantly easier to do when you have data set up in subcollections.

And from their summary:

> So when should you store things in a separate top level collection vs. using subcollections? If you think you **have a situation where you're mostly going to be querying documents based on a common "parent" and only occasionally want to perform queries across all collections, go with a subcollection setup** and enable collection group queries when appropriate. On the other hand, if it seems like no matter how you divide up your documents, the majority of your queries are going to require a collection group query, maybe keep them as a top level collection.

So yeah, sub-collection pattern sounds good!

Reply:
(captured this in https://github.com/sparkletown/internal-sparkle-issues/issues/127#issuecomment-797160724 for future reference)